src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/InsertOneCommandResolver.java

```diff
@@ -34,7 +34,8 @@ public Operation resolveCommand(CommandContext ctx, InsertOneCommand command) {
             null,
             ctx.indexingProjector(),
             command.getClass().getSimpleName(),
-            ctx.collectionSettings());
-    return new InsertOperation(ctx, shreddedDocument);
+            ctx.collectionSettings(),
+            null);
+    return InsertOperation.create(ctx, shreddedDocument);
```

Comment:
Are we not returning document id in case of insertOne

Reply:
No, because "returnDocumentResponses" is only added for "insertMany"?
So in case of failure we do not need `DocumentId`? 

Although maybe that could be a future improvement.